src/payments/search_builder.rs

```diff
@@ -0,0 +1,122 @@
+use async_stream::stream;
+use reqwest::Method;
+use std::pin::Pin;
+use tokio_stream::Stream;
+
+use super::{
+    get_builder::PaymentGetBuilder,
+    types::{PartialPaymentResult, PaymentResponse, PaymentSearchOptions, PaymentSearchResponse},
+};
+use crate::{
+    client::MercadoPagoClient,
+    common::{resolve_json, MercadoPagoRequestError},
+};
+
+/// Builder for search payments
+///
+/// # Arguments
+///
+/// * `options` - Options to search for payments
+///
+/// # Example
+/// ```
+/// PaymentSearchBuilder(
+///     PaymentSearchOptions {
+///         limit: Some(10),
+///         sort: Some(PaymentSearchSort::DateLastUpdated)
+///         ..Default::default()
+///     }
+/// )
+/// ```
+///
+/// # Docs
+/// <https://www.mercadopago.com.br/developers/pt/reference/payments/_payments_search/get>
+pub struct PaymentSearchBuilder(pub PaymentSearchOptions);
+
+impl PaymentSearchBuilder {
+    /// Send the request
```

Comment:
Essa documentação ta errada


```suggestion
    /// This function creates a stream of payments, it goes through all the pages.
    /// When you fetch a payment, it will check if you reached the end of a page, if you have, it will fetch another page and returno the first payment on that page, other wise it gives you the next payment from the current page
```